lib/galaxy/datatypes/gsas2.py

```diff
@@ -0,0 +1,117 @@
+"""
+classes of datatypes from GSASII
+"""
+
+import logging
+
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.data import Text
+
+log = logging.getLogger(__name__)
+
+
+class Gsas2Project(Binary):
+    """
+    GSASII .gpx project file format
+    file is pickled can be read by pickle.load(fp,encoding='latin-1')
```

Comment:
Could we unpickle and use in the sniffer? Or is there not a clear identifier even when unpickled?

Reply:
We can unpickle it, but I believe we need the GSASII project class which requires importing GSASII modules. Once unpickled they should be indentifiable. Is this something we should do?

Reply:
Ah OK, based on the docstring I thought it just required the stdlib `pickle` library - if it requires a specific import it's probably not worth it unless we a good reason, like wanting to extract and set metadata for the file to show in the UI. We did this for the Atomic Simulation Environment for some of the atomic structure file formats, but even then it's an optional import that fails gracefully if not installed.